src/pip/_internal/resolution/resolvelib/candidates.py

```diff
@@ -93,16 +90,17 @@ def make_install_req_from_editable(
     )
 
 
-def make_install_req_from_dist(
-    dist: Distribution, template: InstallRequirement
+def _make_install_req_from_dist(
+    dist: BaseDistribution, template: InstallRequirement
 ) -> InstallRequirement:
-    project_name = canonicalize_name(dist.project_name)
+    from pip._internal.metadata.pkg_resources import Distribution as _CompatDist
```

Comment:
This name is confusing. :/

`PkgResourcesDistribution` is longer, but would be clearer.

Reply:
This is intended to be temporary so I didn’t really think too hard about the name. But yeah it should at least be consisteny…